From 0b40c9d23dc111b8b40cdfece166fefdfcda44bc Mon Sep 17 00:00:00 2001
From: Tomas Barton <barton.tomas@gmail.com>
Date: Thu, 25 Aug 2022 23:32:41 +0200
Subject: [PATCH] Cleanup table entry (#17) (#18)

* Cleanup table entry (#17)
* Fix indentation, avoid code duplication
* Move comment

(cherry picked from commit 5b4712121b4f42fe853e075fc211723f7463c2b4)
---
 iptstate.cc | 25 +++++++++++++++----------
 1 file changed, 15 insertions(+), 10 deletions(-)

diff --git a/iptstate.cc b/iptstate.cc
index 9beff1d9b2fe7..2d072ba3655f9 100644
--- a/iptstate.cc
+++ b/iptstate.cc
@@ -1228,6 +1228,18 @@ int conntrack_hook(enum nf_conntrack_msg_type nf_type, struct nf_conntrack *ct,
   return NFCT_CB_CONTINUE;
 }
 
+/**
+ * Nuke the tentry_t's we made before deleting the vector of pointers
+ */
+void clear_table(vector<tentry_t*> &stable)
+{
+  for (tentry_t* entry: stable) {
+    delete entry;
+  }
+  stable.clear();
+}
+
+
 /*
  * This is the core of this program - build a table of states.
  *
@@ -1259,15 +1271,8 @@ void build_table(flags_t &flags, const filters_t &filters, vector<tentry_t*>
   /*
    * Initialization
    */
-  // Nuke the tentry_t's we made before deleting the vector of pointers
-  for (
-      vector<tentry_t*>::iterator it = stable.begin();
-      it != stable.end();
-      it++
-  ) {
-    delete *it;
-  }
-  stable.clear();
+  clear_table(stable);
+
   counts.tcp = counts.udp = counts.icmp = counts.other = counts.skipped = 0;
 
   cth = nfct_open(CONNTRACK, 0);
@@ -2258,7 +2263,6 @@ void interactive_help(const string &sorting, const flags_t &flags,
   halfdelay(1);
 }
 
-
 /*
  * MAIN
  */
@@ -2958,6 +2962,7 @@ int main(int argc, char *argv[])
   } // end while(1)
 
   out:
+  clear_table(stable);
 
   /*
    * The user has broken out of the loop, take down the curses
